### ISSUE TITLE
numpad on macOS eventually stops inputting numbers

### ISSUE THREAD

**Issue by toadgee** [fish-shell/fish-shell#7290]
tldr; on macOS the numeric keypad ends up sending key codes which are interpreted incorrectly.

fish 3.1.2
macOS 10.15.5 Darwin toddha-dev16 19.5.0 Darwin Kernel Version 19.5.0: Tue May 26 20:41:44 PDT 2020; root:xnu-6153.121.2~2/RELEASE_X86_64 x86_64
xterm-256color

Apologies in advance; I don't know if this is a fish issue or a Terminal.app issue, or something related to my setup (something not cleaning up itself properly). I did a search of existing issues and nothing "popped" although it could be my search query was not quite correct.

I have a "full sized" macOS keyboard with the numpad. What happens is that when I type a key in the numpad (0-9) i get the incorrect characters inserted in Terminal.
return - "OM"

- "+" : "Ol"
- "-" : "Om"
- "." : "On"
- "0" : "Op"
- "1" : "Oq"
- "2" : "Or"
- ...
- "9" : "Oy"

=/*  - all work fine

https://asciinema.org/a/sAdBBknpfEOHclSLQjKC3yebN

I'm running (nearly) the latest macOS - though have seen this problem quite consistently since probably fish 3.0 on ... 10.15.0? I've also seen it on a couple different machines (which have generally the same setup). I generally leave my terminal windows open for quite a while (days, weeks) and happen to notice it in certain Terminal tabs. If i create a new tab, then the problem doesn't carry on to the new tab (i.e. process) - but still remains in the old tab. Whenever I encounter this I just kill the tab and create a new tab.

Interestingly, if I execute "fish" within that busted window, it is still broken, leading me to suspect that it's something in my environment that causing havoc.

Output from set (scrubbed of some domain specific settings and fish_color):

CMD_DURATION 1
COLUMNS 160
FISH_VERSION 3.1.2
HOME /var/folders/3q/jntmqb557l9gm705nfdmwn280000gn/T/tmp.paKBX95B
IFS \n\ \t
LANG en_US.UTF-8
LINES 53
LOGNAME toddha
PATH '/usr/local/bin'  '/usr/bin'  '/bin'  '/usr/sbin'  '/sbin'  …
PWD /Volumes/Source/depot/devmain/dev/mbu/projects
SHELL /usr/local/bin/fish
SHLVL 4
SSH_AUTH_SOCK /private/tmp/com.apple.launchd.P6uZYdeQM9/Listeners
TEMP /Volumes/Source/depot/temp/
TERM xterm-256color
TERM_PROGRAM Apple_Terminal
TERM_PROGRAM_VERSION 433
TERM_SESSION_ID 9C9EAA79-E4EC-485B-B26F-796652F33CC7
TMPDIR /Volumes/Source/depot/temp/
USER toddha
XPC_FLAGS 0x0
XPC_SERVICE_NAME 0
_ set
___fish_git_prompt_status_order 'stagedstate'  'invalidstate'  'dirtystate'  'untrackedfiles'
__fish_active_key_bindings fish_default_key_bindings
__fish_added_user_paths
__fish_bin_dir /usr/local/bin
__fish_config_dir /var/folders/3q/jntmqb557l9gm705nfdmwn280000gn/T/tmp.paKBX95…
__fish_config_interactive_done
__fish_data_dir /usr/local/share/fish
__fish_help_dir /usr/local/share/doc/fish
__fish_initialized 3100
__fish_sysconf_dir /usr/local/etc/fish
__fish_user_data_dir /var/folders/3q/jntmqb557l9gm705nfdmwn280000gn/T/tmp.paKBX95…
fish_bind_mode default
fish_complete_path '/var/folders/3q/jntmqb557l9gm705nfdmwn280000gn/T/tmp.paKBX9…
fish_function_path '/var/folders/3q/jntmqb557l9gm705nfdmwn280000gn/T/tmp.paKBX9…
fish_greeting Welcome\ to\ fish,\ the\ friendly\ interactive\ shell\nType\…
fish_key_bindings fish_default_key_bindings
fish_pid 4290
fish_private_mode
fish_prompt_hg_status_added ✚
fish_prompt_hg_status_copied ⇒
fish_prompt_hg_status_deleted ✖
fish_prompt_hg_status_modified '*'
fish_prompt_hg_status_order 'added'  'modified'  'copied'  'deleted'  'untracked'  'unme…
fish_prompt_hg_status_unmerged !
fish_prompt_hg_status_untracked '?'
history set fish exit 'asciinema rec' root exit
hostname toddha-dev16
pipestatus 0
status 0
umask 0022
version 3.1.2

I'll do my best to keep my busted tab up if anybody has any ideas...

**Comment by zanchey** [comment]
Something is putting your terminal into [application keypad mode](https://vt100.net/docs/vt102-ug/chapter4.html#S4.4). `tput rmkx` should put it back. fish did play with these settings briefly (#2139, #2309), but shouldn't do at present.

**Comment by faho** [comment]
The question appears to be answered - you have something that changes your terminal settings without changing them back, find it and change it.

(fish should probably still do this itself, but that's #2139)

**Comment by toadgee** [comment]
Doh, thanks @zanchey !

I'll track that down... (my build system at work is overly complex, so it's likely because I've cmd+c'd out of it at some point and it didn't get to clean up). I can probably fix this by adding something to my fish config to magically reset it.

Cheers, appreciate the support.
